From b48a2dfae26fa3b4af8e65fb5953b3caf62c137b Mon Sep 17 00:00:00 2001
From: Paresh Chaudhary <paresh.chaudhary@rockwellcollins.com>
Date: Mon, 21 May 2018 14:34:23 -0500
Subject: [PATCH] checksec: Fixed issue with relative path

Before this patch script was not able find exists directory when user pass
relative directory path with '--dir' or '-d' option and also we faced this error
when we execute script with relative path.

This patch fixed  "No such file or directory" error in both cases.

https://github.com/slimm609/checksec.sh/issues/54

Signed-off-by: Paresh Chaudhary <paresh.chaudhary@rockwellcollins.com>
---
 checksec | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/checksec b/checksec
index 24b521f..baf8d63 100755
--- a/checksec
+++ b/checksec
@@ -1193,7 +1193,7 @@ do
     echo_message "RELRO           STACK CANARY      NX            PIE             RPATH      RUNPATH      FORTIFY Checked         Total   Filename\n" '' "<dir name='$tempdir'>\n" "{ \"dir\": { \"name\":\"$tempdir\" },"
     fdircount=0
     fdirtotal=0
-    for N in $(find $tempdir -type f); do
+    for N in $(find . -type f); do
       if [[ "$N" != "[A-Za-z1-0]*" ]]; then
         out=$(file "$N")
         if [[  $out =~ ELF ]] ; then
@@ -1201,7 +1201,7 @@ do
         fi
       fi
     done
-    for N in $(find $tempdir -type f); do
+    for N in $(find . -type f); do
       if [[ "$N" != "[A-Za-z1-0]*" ]]; then
     # read permissions?
     if [[ ! -r "$N" ]]; then
-- 
1.9.1

